### PR TITLE
Fix & migrate user_id column type to bigint

### LINKED ISSUE #54214: Fix & migrate user_id column type to bigint
**What happened**:
After migrating to Grafana 9.x  - there was a new table added for query history "query_history_star" and now this is causing an issue when adding/deleting a record from/into it. 

**Observation**:
The user_id there is of type INTEGER, but the real user id is of type BIGINT and this causing one sql error. 
```value is out of range for type integer = (MISSING)```

**Environment**:
- Grafana version: 9.0.2
- Data source type & version: N/A
- OS Grafana is installed on: Windows/Linux
- User OS & Browser: All
- Grafana plugins: N/A
- Others: N/A

### PR DESCRIPTION
**Pull request by Bujupah**
**What this PR does / why we need it**:
This will fix columns type in **query_history_star**, **query_history** and **short_url** table for `user_id` column

**Which issue(s) this PR fixes**:
Fixes #54214

### CODE REVIEW

**Comment by Bujupah** [comment]
Will probably need to backport this too all 9.x versions

File: pkg/services/sqlstore/migrations/query_history_star_mig.go [hunk 1]
```diff
@@ -24,4 +24,7 @@ func addQueryHistoryStarMigrations(mg *Migrator) {
 	mg.AddMigration("add column org_id in query_history_star", NewAddColumnMigration(queryHistoryStarV1, &Column{
 		Name: "org_id", Type: DB_BigInt, Nullable: false, Default: "1",
 	}))
+	
+	mg.AddMigration("alter table query_history_star_mig column user_id type to bigint", NewRawSQLMigration("ALTER TABLE query_history_star ALTER COLUMN user_id TYPE BIGINT;"))
```
Comment (ifrost) [inline pkg/services/sqlstore/migrations/query_history_star_mig.go:28]: @grafana/backend-platform should we use this instead?

```suggestion
	mg.AddMigration("alter table query_history_star_mig column user_id type to bigint", NewRawSQLMigration("").Mysql("ALTER TABLE query_history_star ALTER COLUMN user_id TYPE BIGINT;"))
```

Comment (sakjur) [reply pkg/services/sqlstore/migrations/query_history_star_mig.go:28]: You'll need to make sure you get the database into the state you want it to be in for MySQL, Postgres, and SQLite. For MySQL, the format of the migration needs to be different from Postgres.

It'll probably look something like this if you don't need a migration for SQLite as well.

```suggestion
	mg.AddMigration("alter table query_history_star_mig column user_id type to bigint", NewRawSQLMigration("").Mysql("ALTER TABLE query_history_star MODIFY user_id BIGINT;")).Postgres("ALTER TABLE query_history_star ALTER COLUMN user_id TYPE BIGINT;"))
```

**Review by ifrost — APPROVED** [review]
LGTM 👍 any other concerns @grafana/backend-platform? 

@Bujupah  - the build is failing at linting stage. I think it might be because of trailing spaces/tabs. Could you run `gofmt -w ./pkg/services/sqlstore/migrations/query_history_star_mig.go`?

File: pkg/services/sqlstore/migrations/query_history_star_mig.go [hunk 2]
```diff
@@ -24,4 +24,7 @@ func addQueryHistoryStarMigrations(mg *Migrator) {
 	mg.AddMigration("add column org_id in query_history_star", NewAddColumnMigration(queryHistoryStarV1, &Column{
 		Name: "org_id", Type: DB_BigInt, Nullable: false, Default: "1",
 	}))
+	
+	mg.AddMigration("alter table query_history_star_mig column user_id type to bigint", NewRawSQLMigration("").Mysql("ALTER TABLE query_history_star MODIFY user_id BIGINT;").Postgres("ALTER TABLE query_history_star ALTER COLUMN user_id TYPE BIGINT;"))
```
Comment (Bujupah) [inline pkg/services/sqlstore/migrations/query_history_star_mig.go:28]:
```suggestion
	mg.AddMigration("alter table query_history_star_mig column user_id type to bigint",
		NewRawSQLMigration("").
			Mysql("ALTER TABLE query_history_star MODIFY user_id BIGINT;").
			Postgres("ALTER TABLE query_history_star ALTER COLUMN user_id TYPE BIGINT;"))
```

**Comment by yusuf-multhan** [comment]
This will probably need change in query_history_mig.go as well

`mg.AddMigration("alter column created_by type to bigint", NewRawSQLMigration("ALTER TABLE query_history ALTER COLUMN created_by TYPE BIGINT;"))`

**Comment by Bujupah** [comment]
Still having this

```
+ make lint-go
(re)installing /go/bin/golangci-lint-v1.40.1
lint via golangci-lint
/go/bin/golangci-lint-v1.40.1 run \
	--config ./conf/.golangci.toml \
	./pkg/...
pkg/services/sqlstore/migrations/query_history_star_mig.go:27: File is not `goimports`-ed (goimports)
	
make: *** [golangci-lint] Error 1
Makefile:135: recipe for target 'golangci-lint' failed

```

**Comment by ifrost** [comment]
> Still having this
> 
> ```
> + make lint-go
> (re)installing /go/bin/golangci-lint-v1.40.1
> lint via golangci-lint
> /go/bin/golangci-lint-v1.40.1 run \
> 	--config ./conf/.golangci.toml \
> 	./pkg/...
> pkg/services/sqlstore/migrations/query_history_star_mig.go:27: File is not `goimports`-ed (goimports)
> 	
> make: *** [golangci-lint] Error 1
> Makefile:135: recipe for target 'golangci-lint' failed
> ```

I think extra tabs in empty lines are the problem, not sure why `gofmt` does it remove them:

Screen Shot 2022-09-02 at 13 43 01

File: pkg/services/sqlstore/migrations/query_history_star_mig.go [hunk 3]
```diff
@@ -24,4 +24,10 @@ func addQueryHistoryStarMigrations(mg *Migrator) {
 	mg.AddMigration("add column org_id in query_history_star", NewAddColumnMigration(queryHistoryStarV1, &Column{
 		Name: "org_id", Type: DB_BigInt, Nullable: false, Default: "1",
 	}))
+	
+	mg.AddMigration("alter table query_history_star_mig column user_id type to bigint",
+		NewRawSQLMigration("").
+			Mysql("ALTER TABLE query_history_star MODIFY user_id BIGINT;").
+			Postgres("ALTER TABLE query_history_star ALTER COLUMN user_id TYPE BIGINT;"))
+
```
Comment (Bujupah) [inline pkg/services/sqlstore/migrations/query_history_star_mig.go:32]:
```suggestion

```

File: pkg/services/sqlstore/migrations/query_history_star_mig.go [hunk 4]
```diff
@@ -24,4 +24,9 @@ func addQueryHistoryStarMigrations(mg *Migrator) {
 	mg.AddMigration("add column org_id in query_history_star", NewAddColumnMigration(queryHistoryStarV1, &Column{
 		Name: "org_id", Type: DB_BigInt, Nullable: false, Default: "1",
 	}))
+
```
Comment (Bujupah) [inline pkg/services/sqlstore/migrations/query_history_star_mig.go:27]:
```suggestion

```

**Comment by Bujupah** [comment]
This is giving me hard time guys 🙄
My editor's prettier is adding one backspace automatically

**Comment by Bujupah** [comment]
Like @yusuf-multhan mentioned. that mistake is there too...
Also seeing same in `short_url` table for `created_by` column.

Shall I raise that in same PR ?

**Comment by ifrost** [comment]
> Like @yusuf-multhan mentioned. that mistake is there too... Also seeing same in `short_url` table for `created_by` column.
> 
> Shall I raise that in same PR ?

I think it can be added to the same PR, and you can just change the PR description. Is there an issue reported for `short_url`? Just to cross-reference it with this PR.

**Comment by ifrost** [comment]
It keeps failing at linting. I'll open a mirror PR for it https://github.com/grafana/grafana/pull/55084

**Comment by ifrost** [comment]
Closing as it was added in #55084

**Pull request closed without merging**